database/user_database.go

```diff
@@ -107,22 +145,50 @@ func (u RedisDatabase) PopBuy(user string) (stock string, cost decimal.Decimal,
 
 // AddFunds adds amount dollars to the user account
 func (u RedisDatabase) AddFunds(user string, amount decimal.Decimal) error {
-	fmt.Print(user)
-	fmt.Print(amount)
+	conn := u.getConn()
+	_, err := conn.Do("INCRBYFLOAT", user+":Balance", amount)
+	if err != nil {
+		panic(err)
+	}
+	conn.Close()
 	return nil
 }
 
 // GetFunds returns the amount of available funds in a users account
 func (u RedisDatabase) GetFunds(user string) (decimal.Decimal, error) {
-	return decimal.NewFromFloat(0), nil
+	conn := u.getConn()
+	r, err := redis.String(conn.Do("GET", user+":Balance"))
+	if err != nil {
+		panic(err)
+	}
+	conn.Close()
+	receivedValue, err := decimal.NewFromString(r)
+	return receivedValue, err
 }
 
 // RemoveFunds remove n funds from the user's account
 func (u RedisDatabase) RemoveFunds(user string, amount decimal.Decimal) error {
+	conn := u.getConn()
+	_, err := conn.Do("INCRBYFLOAT", user+":Balance", amount.Neg())
+	if err != nil {
+		panic(err)
```

Comment:
So either we can return the err, the server will return a failure (-1) and continue recieving requests. Or we can panic and the server will shutdown (i think). Wish makes sense if the database is failing we probably don't wanna continue. Idk up to you?

Reply:
I say return the err -- I'll make those changes